src/main/java/uk/gov/ons/fwmt/job_service/rest/impl/RestHelper.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.ons.fwmt.job_service.rest.impl;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import uk.gov.ons.fwmt.job_service.exceptions.types.ResourceServiceInaccessibleException;
+import uk.gov.ons.fwmt.job_service.exceptions.types.ResourceServiceMalfunctionException;
+
+import java.util.Optional;
+
+@Slf4j
+public class RestHelper {
+  public static <T> Optional<T> get(RestTemplate restTemplate, String url, Class<T> klass, Object... parameters) {
+    try {
+      final ResponseEntity<T> responseEntity = restTemplate.getForEntity(url, klass, parameters);
+      if (responseEntity.getStatusCode().is2xxSuccessful()) {
+        // return a successful result
+        log.info(String.format("GET successful, received %s", klass.getSimpleName()));
+        return Optional.ofNullable(responseEntity.getBody());
+      }
+      // any success that doesn't have a 200 is an unexpected error
+      ResourceServiceMalfunctionException malfunctionException = new ResourceServiceMalfunctionException(
+          String.format("Unexpected HTTP code: %d", responseEntity.getStatusCode().value()));
+      log.info("GET failed with non-2xx status code");
```

Comment:
log the failed code with class parameters.
This becomes a duplicate info if the Error also logs same info